### PR TITLE
fix: infinite query glitch on invalid spec

### PR DESCRIPTION
**Pull request by djbarnwal**
Fixes

https://github.com/rilldata/rill-private-issues/issues/630
https://github.com/rilldata/rill-private-issues/issues/631

Instead of infinite loading state, we show the error message for the failed request

image

### CODE REVIEW

File: web-common/src/features/charts/prompt/ChartDataDisplay.svelte
Comment (ericpgreen2) [inline web-common/src/features/charts/prompt/ChartDataDisplay.svelte:1]: Now I'm not following what the `prompt` directory is supposed to represent. I thought it was supposed to encapsulate all the code related to AI chart generation?

**Review by ericpgreen2 — COMMENTED** [review]
UXQA: when I edit values in the `data:` section, the chart does not respond. [See JAM](https://jam.dev/c/8746c49b-36f4-4539-bd85-c91ae395d6c5).

File: web-common/src/features/charts/prompt/ChartDataDisplay.svelte
Comment (djbarnwal) [reply web-common/src/features/charts/prompt/ChartDataDisplay.svelte:1]: Don't have much context on this. Probably @AdityaHegde @briangregoryholmes would know more.

**Comment by djbarnwal** [comment]
> UXQA: when I edit values in the `data:` section, the chart does not respond. [See JAM](https://jam.dev/c/8746c49b-36f4-4539-bd85-c91ae395d6c5).

Did a bit of digging. The issue is also on main. The `resource` query needs to be invalidated when the data query is changing. Will create a separate PR for that.

File: web-common/src/features/charts/prompt/ChartDataDisplay.svelte
Comment (AdityaHegde) [reply web-common/src/features/charts/prompt/ChartDataDisplay.svelte:1]: It is. I think just this file is misplaced.

**Review by ericpgreen2 — APPROVED** [review]
Approving, just it'd be nice to move the non-AI code outside of the `charts/prompt` directory before merging.

**Comment by djbarnwal** [comment]
> Approving, just it'd be nice to move the non-AI code outside of the `charts/prompt` directory before merging.

The whole custom dashboard / charts terminology needs to be updated to canvas and components. Will do it in a separate PR.

**Pull request merged**